### ISSUE TITLE
ShouldBeOfType<> fails on collections.

### ISSUE THREAD

**Issue by godrose** [shouldly/shouldly#373]
Repro steps:

1. Have an object that is of IEnumerable<IService> type.
2. Call ShouldBeOfType<IEnumerable<IService>> on the object.

Expected: returns true;
Actual: returns false with the following message:
"System.Collections.Generic.IEnumerable`1[IService]
    should be of type
System.Collections.Generic.IEnumerable`1[IService]
    but was
System.RuntimeType"

Note: The object's type is constructed using ``IEnumerable<>.MakeGeneric(typeof(IService``. Yes, it's necessary.

**Comment by JosephWoodward** [comment]
Thanks for raising this. I'll see if I can reproduce it.

**Comment by JosephWoodward** [comment]
Looks like the following also fails:

    IEnumerable<string> temp = Enumerable.Empty<string>();
    temp.ShouldBeOfType<IEnumerable<string>>();

with the following error message:

    []
        should be of type
    System.Collections.Generic.IEnumerable`1[System.String]
        but was
    System.String[]`

**Comment by ssiko** [comment]
@JosephWoodward Might we want to use `ShouldBeAssignableTo<T>` in these scenarios? We are using `GetType()` in `ShouldBeOfType<T>`, this will return the Concrete Class' Type for the instance, not its interfaces. In order to assert that a concrete class is implementing a specific interface `ShouldBeAssignableTo<T>` should do the trick...Let me know your thoughts.

```cs
            IEnumberable<string> strings = new List<string>();
            //passes
            strings.ShouldBeAssignableTo<IEnumerable<string>>();
            //passes
            strings.ShouldBeAssignableTo<IList<string>>();
            //fails
            strings.ShouldBeOfType<IEnumerable<string>>(); 
```

**Comment by JosephWoodward** [comment]
> @JosephWoodward Might we want to use `ShouldBeAssignableTo<T>` in these scenarios? We are using `GetType()` in `ShouldBeOfType<T>`, this will return the Concrete Class' Type for the instance, not its interfaces. In order to assert that a concrete class is implementing a specific interface `ShouldBeAssignableTo<T>` should do the trick...Let me know your thoughts.
> 
> ```cs
>             IEnumberable<string> strings = new List<string>();
>             //passes
>             strings.ShouldBeAssignableTo<IEnumerable<string>>();
>             //passes
>             strings.ShouldBeAssignableTo<IList<string>>();
>             //fails
>             strings.ShouldBeOfType<IEnumerable<string>>(); 
> ```

You're right `ShouldBeAssignableTo<T>` is the right assertion to use. I'm going to close this issue.
